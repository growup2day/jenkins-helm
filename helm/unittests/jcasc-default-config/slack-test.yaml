templates:
  - config-jcasc.yaml
tests:
  - it: should include slack config in default config scripts when configured
    set:
      JCasC:
        defaultConfig: true
        slackNotifier:
          credentialId: "slack-cred"
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-slack.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-slack.yaml]
  - it: should include blank slack config in default config scripts when not configured
    set:
      JCasC:
        defaultConfig: true
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-slack.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-slack.yaml]
  - it: should not include slack config in default config scripts when there is custom config script for the section
    set:
      JCasC:
        defaultConfig: true
        configScripts:
          custom-override: |
            unclassified:
              slackNotifier:
                teamDomain: ""
    asserts:
      - isNull:
          path: data.[jcasc-default-config-slack.yaml]
      - isNotNull:
          path: data.[custom-override.yaml]
