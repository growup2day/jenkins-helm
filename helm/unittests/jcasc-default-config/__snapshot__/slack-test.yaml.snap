should include blank slack config in default config scripts when not configured:
  1: |
    |-
      unclassified:
        # Configure System > Slack
        slackNotifier:
          teamDomain: ""
          tokenCredentialId: ""
should include slack config in default config scripts when configured:
  1: |
    |-
      unclassified:
        # Configure System > Slack
        slackNotifier:
          botUser: false
          sendAsText: false
          teamDomain: "anznz"
          tokenCredentialId: "slack-cred"
